Session.js

```diff
@@ -0,0 +1,53 @@
+class Session { 
+// To be utilized when unwrapping JSON from the future servlets. Will be on the 
+// client side and be used as a buffer for client to server information needed  
+// for other components of Virtual Movie Night.
+ 
+    constructor() {
+        this.sessionId = expected.sessionId;
+        this.ipOfVM = expected.ipOfVM;
+        this.listOfAttendees = expected.listOfAttendees;
+        this.screenNameOfController = expected.screenNameOfController; 
+    }
+    
+    setSessionId(id) { 
+        this.sessionId = id;
+    }
+
+    setIpOfVM(ip) {
+        this.ipOfVM = ip;
+    }
+
+    setListOfAttendees(attendees) {
+        this.listOfAttendees = attendees;
+    }
+
+    setScreenNameOfController(controller) {
+        this.screenNameOfController = controller;
+    }
+
+    getSessionId() {
+        return this.sessionId;
+    } 
+
+    getIpOfVM() {
+        return this.ipOfVM;
+    } 
+
+    getListOfAttendees() { 
+       return this.listOfAttendees;    
+    }
+    
+    getScreenNameOfController() {
+       return this.screenNameOfController; 
+    }
+    
+    static fromObject(obj) {  
```

Comment:
you have to create the Session object first and then call the setters from the new object.

Reply:
you don’t want to overwrite obj in line 45

static fromObject(obj) {
const session = new Session();
session.setSessionId(...)
...
return session
}

Reply:
you don’t want to overwrite obj in line 45

static fromObject(obj) {
const session = new Session();
session.setSessionId(...)
...
return session
}